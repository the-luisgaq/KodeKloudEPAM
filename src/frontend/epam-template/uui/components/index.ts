export * from './buttons';
export * from './datePickers';
export * from './dnd';
export * from './inputs';
export * from './layout';
export * from './overlays';
export * from './pickers';
export * from './typography';
export * from './widgets';
export * from './tables';
export * from './filters';
export * from './navigation';
export * from './forms';
export * from './fileUpload';
export * from './errors';
