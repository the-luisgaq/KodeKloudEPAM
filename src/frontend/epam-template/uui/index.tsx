import './assets/styles/typography.scss';

export * from './components';
export * from './i18n';
export * from './helpers';
export * from './settings';

export type { DataTableCellContainerProps } from '@epam/uui-components';
export { DataTableCellContainer } from '@epam/uui-components';
