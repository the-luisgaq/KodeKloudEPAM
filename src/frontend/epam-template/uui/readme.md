# UUI package

This package is a part of [EPAM UUI](https://github.com/epam/UUI) library.

`@epam/uui` contains theme agnostic set of UUI components. 

It's required to connect any theme to your application for components styling. 
You can read more about UUI Themes [here](https://uui.epam.com/documents?id=overview-themes&mode=doc&isSkin=null&category=themes).

Full set of components can be found on [UUI site](https://uui.epam.com/documents?category=components&id=accordion&mode=doc).