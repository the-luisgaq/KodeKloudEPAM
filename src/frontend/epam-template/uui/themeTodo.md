# TODO
- [ ] In FilterRangeDatePickerBody.tsx (in 'promo' now) we would like to use RangeDatePicker component in footer, and get rid of copy-paste.
- [ ] Remade SearchInput in PickerInput to fit Figma UUI3 styles.
- [ ] PgeButton (Paginator) colors in figma and in our site are different. Where are the correct set?

- [ ] WithMods function - need to rework:
     1) case when we need to redefine some variable typo, for example - color.
     2) case when we need to add some specific for skin typing
     3) we removed inherited IHasCX, because DatePicker no need cx props, but it has inputCx and BodyCx, so we need to remade it
- [ ] подумать нужен ли нам Text компонент в UUI
- [ ] NotificationCard: we need to remove the semantic varieties of notificationCards from skins and re-export them from uui when we get rid of hanging the theme directly in skins
- [ ] IconContainer: remove colors for Promo will be - semantic colors + default + secondary, for Loveship - semantic + 'night400'(used in DataPickerRow) + 'default' + 'secondary'

# TODO for UUI package
### Panel
- [ ] discuss background props
### PickerInput
- [ ] DataPickerBody renderNotFound promo has been modified with different logic than loveship
### MainMenu
- [ ] Discuss mainMenu tokens, from whom should we inherit menu tokens?

