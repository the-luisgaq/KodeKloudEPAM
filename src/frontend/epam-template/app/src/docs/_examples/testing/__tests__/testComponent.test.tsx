/**
 *  Note: This example uses  "@epam/uui-test-utils" package to simplify testing.
 */
import React, { useCallback, useState } from 'react';
import {
    setupComponentForTest,
    screen,
    fireEvent,
    renderSnapshotWithContextAsync,
    renderHookWithContextAsync,
    waitFor, renderWithContextAsync,
    act,
} from '@epam/uui-test-utils';
import { TextInput } from '@epam/uui';

/** ****************** */
/** * TestComponent ** */
/** ****************** */

/** Start: This is some component which we are going to test. It's just an example. */
interface TestComponentProps {
    value?: string;
    onValueChange?: (value: string) => void;
}
function TestComponent(props: TestComponentProps) {
    return (
        <TextInput value={ props.value } onValueChange={ props.onValueChange } />
    );
}
/** End */

/**
 *  In this specific case, the "setupComponentForTest" method simplifies testing of next use cases:
 *  - It wraps the component in UUI context;
 *  - It makes it possible to change component's props from the outside without unmounting the component;
 *  - It supports testing of "on-change" workflow, when a callback prop (e.g. onValueChange) updates some other props (e.g. value).
 */
async function setupTestComponent(params: Partial<TestComponentProps>) {
    const { mocks, setProps } = await setupComponentForTest<TestComponentProps>(
        (context) => ({
            value: params.value,
            onValueChange: jest.fn().mockImplementation((newValue) => {
                context.current.setProperty('value', newValue);
            }),
        }),
        (props) => <TestComponent { ...props } />,
    );
    const input = screen.queryByRole('textbox') as HTMLInputElement;
    const dom = { input };
    return {
        setProps,
        mocks,
        dom,
    };
}

describe('TestComponent', () => {
    it('should render with minimum props', async () => {
        const tree = await renderSnapshotWithContextAsync(<TestComponent />);
        expect(tree).toMatchSnapshot();
    });

    it('should render with maximum props', async () => {
        const tree = await renderSnapshotWithContextAsync(<TestComponent value="monday" onValueChange={ jest.fn() } />);
        expect(tree).toMatchSnapshot();
    });

    it('should invoke onValuesChange when user types new value', async () => {
        const { mocks, dom } = await setupTestComponent({ value: 'monday' });
        expect(dom.input.value).toEqual('monday');
        fireEvent.change(dom.input, { target: { value: 'friday' } });
        expect(dom.input.value).toEqual('friday');
        expect(mocks.onValueChange).toHaveBeenLastCalledWith('friday');
    });

    it('should display new value when value property changes', async () => {
        const { setProps, mocks, dom } = await setupTestComponent({ value: 'monday' });
        setProps({ value: 'tuesday' });
        expect(dom.input.value).toEqual('tuesday');
        expect(mocks.onValueChange).not.toHaveBeenCalled();
    });
});

describe('TestComponent (without setupComponentForTest)', () => {
    it('should invoke onValuesChange when user types new value', async () => {
        const onValueChangeMock = jest.fn();
        await renderWithContextAsync(<TestComponent value="monday" onValueChange={ onValueChangeMock } />);
        const input = screen.queryByRole('textbox') as HTMLInputElement;
        expect(input.value).toEqual('monday');
        fireEvent.change(input, { target: { value: 'friday' } });
        expect(onValueChangeMock).toHaveBeenLastCalledWith('friday');
    });
});

/** ****************** */
/** * useTestHook ** */
/** ****************** */

/** Start: This is some hook which we are going to test. It's just an example. */
function useTestHook() {
    const [result, setResult] = useState<number>(0);
    const increment = useCallback(() => {
        setResult((prev) => prev + 1);
    }, []);

    return {
        increment,
        result,
    };
}
/** End */

describe('useTestHook', () => {
    it('should increment', async () => {
        const { result } = await renderHookWithContextAsync(useTestHook);
        expect(result.current.result).toBe(0);
        act(() => {
            result.current.increment();
        });
        await waitFor(() => {
            expect(result.current.result).toBe(1);
        });
    });
});
