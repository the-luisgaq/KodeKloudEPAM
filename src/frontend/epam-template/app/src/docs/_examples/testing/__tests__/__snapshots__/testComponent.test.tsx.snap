// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestComponent should render with maximum props 1`] = `
<div
  className="container uui-input-box -clickable root size-36 mode-form"
  onFocus={[Function]}
  tabIndex={-1}
>
  <input
    className="uui-input"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    tabIndex={0}
    type="text"
    value="monday"
  />
</div>
`;

exports[`TestComponent should render with minimum props 1`] = `
<div
  className="container uui-input-box -clickable root size-36 mode-form"
  onFocus={[Function]}
  tabIndex={-1}
>
  <input
    className="uui-input"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    tabIndex={0}
    type="text"
    value=""
  />
</div>
`;
